### DISCUSSION TITLE
Using #each and queryParamRaw when there's only a single queryParam

### DISCUSSION THREAD

**Discussion by jtaylordigital** [mockoon/mockoon#1259] [Help] [answered]
Hi. I'm trying to mock an API that accepts one to many instances of a queryParam called "id". Iterating through this with #each and queryParamRaw works fine if there are 2+ instances of "id", but if there's only one, the "each" section doesn't get executed. Is there a way to make this work regardless of how many ids are present in the call?

Example call:
`/test?id=123&id=456&id=789`

Mock Setup:
```
{
  {{#each (queryParamRaw 'id')}}
    "id": {{this}}
  {{/each}}
}
```

This works as expected and produces this output:
```
{
    "id": 123
    "id": 456
    "id": 789
}
```

However if the call is:
`/test?id=123`
then the output is:
```
{}
```

I was able to get the desired output by adding an "else" into the mock but that feels cumbersome as I have to repeat the mock setup and replace instances of "this."
```
{
  {{#each (queryParamRaw 'id')}}
    "id": {{this}}
  {{else}}
    "id": {{queryParam 'id'}}
  {{/each}}
}
```

**Comment by 255kb** [comment, marked as answer]
Hi, there is no easy solution as the default behavior of the server we are using is to parse single query parameters as strings or as arrays when multiple occurences are found. 
However, you could handle this a bit differently to simplify your code after,  using split/join and a variable:

  ```handlebars
  {{setVar 'ids' (split (join (queryParamRaw 'id') ',') ',')}}
  
  {
    {{#each @ids}}
      "id": {{this}}
    {{/each}}
  }
  ```

**Reply by jtaylordigital** [reply]
That solution works, thanks!
